### DISCUSSION TITLE
[rate-limiting] should not increment on 5xx responses

### DISCUSSION THREAD

**Discussion by ahmadnassri** [Kong/kong#7204] [Ideas and feature requests]
currently rate limiting is blanket across all requests, there should be some sort of method to customize it? perhaps #160 is the answer?

**Comment by thibaultcha** [comment]
I don't see how 160 is related to this? You just want providers to chose a list of status codes (or other) on which not to increment? That would be related to the rate-limiting plugin itself.

**Comment by ahmadnassri** [comment]
> I don't see how 160 is related to this

thinking is: if we do #505, then plugins could be configured per http method, path, etc ... why not response codes as well!

**Comment by thibaultcha** [comment]
The thing is that it makes sense for some plugins like this one, absolutely not for others. It might be allowing too much flexibility at the cost of usability.

**Comment by ahmadnassri** [comment]
yep, which is why its a suggestion ... both approaches are valid (as a plugin specific config or as part of #505)

so if we do decide to response level config in #505, means no need to do plugin level config for this here ...

**Comment by subnetmarco** [comment]
Currently the rate-limiting increments the counters **before** it is being proxied to the upstream service in `access_by_lua`, basically ignoring any status code returned by the upstream service. To implement this feature, the increment needs to happen into `header_filter_by_lua` so that the appropriate check can be implemented.

The schema should support this additional property: `config.ignore_status_codes` which can be an array of status codes.
